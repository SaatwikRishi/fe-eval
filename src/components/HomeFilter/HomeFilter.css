.home-filter {
    display: flex;
    justify-content: space-between;
    min-height: 10vh;
    width: 100%;
    align-items: center;
    margin-top: 30px
    
}

.text-icon {
    font-size: 25px;
}

.filter-text {
    display: flex;
    column-gap: 10px;
}

.filter-text > h3 {
    font-size: 20px;
}

.filter-search > input {
    width: 20vw;
    max-width: 300px;
    height: 40px;
    border: 1px solid ;
    border-radius: 10px;
    font-weight: bolder;
    padding: 0 10px;
    margin-right: 10px ;
}

.filter-search {
    display: flex;
    align-items: center;
}


.search-icon {
    height: 30px;
   
}

.filter-left {
    display: flex;
    flex-direction: column;
    row-gap: 10px;
}
.filter-right {
    display: flex;
    flex-direction: column;
    align-items: end;
    row-gap: 10px;
}