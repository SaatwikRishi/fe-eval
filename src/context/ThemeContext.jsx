import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { makeRequest } from '../util/makeRequest'
import { GET_THEMES } from '../util/constants'
import { useNavigate } from 'react-router-dom'

export const ThemeContext = React.createContext()

export default function ThemeState ({ children }) {
  const [themes, setThemes] = React.useState([])
  const [loading, setLoading] = React.useState(true)
  const [currentTheme, setCurrentTheme] = React.useState(1)
  const getCurrentTheme = () => {
    const th = themes.filter((e) => e.id === currentTheme)[0].colorHexCode
    return th
  }

  const navigate = useNavigate()

  useEffect(() => {
    makeRequest(GET_THEMES())
      .then((e) => {
        setCurrentTheme(e.preferredThemeId)

        setThemes(e.themes)
        setTimeout(() => {
          setLoading(false)
        })
      }, 300)
      .catch((e) => {
        navigate('/error/500')
        setLoading(false)
      })
  }, [])
  if (loading) return <div>Loading</div>
  return (
    <ThemeContext.Provider value={{ themes, setCurrentTheme, getCurrentTheme }}>
      {children}
    </ThemeContext.Provider>
  )
}

ThemeState.propTypes = {
  children: PropTypes.node.isRequired
}
