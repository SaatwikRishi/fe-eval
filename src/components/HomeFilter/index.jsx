import React, { useContext } from 'react'
import RadioInput from '../RadioInput'
import './HomeFilter.css'
import {
  faMagnifyingGlass,
  faChevronUp, faChevronDown,
  faFilter
} from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { ThemeContext } from '../../context/ThemeContext'
import PropTypes from 'prop-types'
export default function HomeFilter ({ onFilterClick }) {
  const [filterOpen, setFilterOpen] = React.useState(false)
  const { getCurrentTheme } = useContext(ThemeContext)
  const [name, setName] = React.useState('')
  const items = [
    ['ALL', 'REGISTERED'],
    ['BOOKMARKED', 'SEATS AVAILABLE']
  ]

  const [selected, setSelected] = React.useState(-1)

  const handleSelectionChange = (item) => {
    setSelected(selected === item ? -1 : item)
    switch (item) {
      case 'ALL':
        onFilterClick('all')
        break
      case 'REGISTERED':
        onFilterClick('isRegistered')
        break
      case 'BOOKMARKED':
        onFilterClick('isBookmarked')
        break
      case 'SEATS AVAILABLE':
        onFilterClick('areSeatsAvailable')
        break
      default:
        break
    }
  }

  return (
    <div className="home-filter">
      <div className="filter-left">
        <div className="filter-text" style={{ color: getCurrentTheme() }} onClick={() => setFilterOpen(!filterOpen)}>
          <FontAwesomeIcon
            className="text-icon"
            icon={faFilter}
            color={getCurrentTheme()}
          />
          <h3>FILTER</h3>
          <FontAwesomeIcon
            className="text-icon"
            color={getCurrentTheme()}
            icon={filterOpen ? faChevronUp : faChevronDown}
          />
        </div>
       {filterOpen && <>
        {items[0].map((item) => (
          <RadioInput
            key={Math.random()}
            name={item}
            onClick={() => handleSelectionChange(item)}
            selected={item === selected}
          />
        ))}
       </>}
      </div>

   {filterOpen && <div className="filter-right">
        <div className="filter-search">
          <input
            type="search"
            onChange={(e) => {
              setName(e.target.value)
            }}
            style={{
              color: getCurrentTheme()
            }}
            placeholder="EVENT NAME"
          />
          <FontAwesomeIcon
            className="search-icon"
            onClick={(e) => onFilterClick('name', name)}
            color={getCurrentTheme()}
            icon={faMagnifyingGlass}
          />
        </div>
        {items[1].map((item) => (
          <RadioInput
            key={Math.random()}
            name={item}
            onClick={() => handleSelectionChange(item)}
            isRight={true}
            selected={item === selected}
          />
        ))}
      </div> }

    </div>
  )
}

HomeFilter.propTypes = {
  onFilterClick: PropTypes.func
}
