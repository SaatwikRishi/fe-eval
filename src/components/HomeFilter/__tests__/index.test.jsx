import HomeFilter from '..'
import { ThemeContext } from '../../../context/ThemeContext'
import React from 'react'
import { render } from '@testing-library/react'

describe('Tests for Home Filter', () => {
  const getCurrentTheme = jest.fn()
  const setCurrentTheme = jest.fn()
  it('should render the component', () => {
    const { container } = render(<ThemeContext.Provider value={{
      setCurrentTheme,
      getCurrentTheme,
      themes: [
        {
          id: 1,
          colorHexCode: '#000000'
        },
        {
          id: 2,
          colorHexCode: '#800080'
        },
        {
          id: 3,
          colorHexCode: '#0000FF'
        },
        {
          id: 4,
          colorHexCode: '#9B9999'
        }
      ]
    }}>
              <HomeFilter />
            </ThemeContext.Provider>)
    expect(container).toMatchSnapshot()
  })
})
