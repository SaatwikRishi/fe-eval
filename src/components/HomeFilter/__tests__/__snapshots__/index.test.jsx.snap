// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Home Filter should render the component 1`] = `
<div>
  <div
    class="home-filter"
  >
    <div
      class="filter-left"
    >
      <div
        class="filter-text"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-filter text-icon"
          data-icon="filter"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.9 54.9C10.5 40.9 24.5 32 40 32H472c15.5 0 29.5 8.9 36.1 22.9s4.6 30.5-5.2 42.5L320 320.9V448c0 12.1-6.8 23.2-17.7 28.6s-23.8 4.3-33.5-3l-64-48c-8.1-6-12.8-15.5-12.8-25.6V320.9L9 97.3C-.7 85.4-2.8 68.8 3.9 54.9z"
            fill="currentColor"
          />
        </svg>
        <h3>
          FILTER
        </h3>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down text-icon"
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
